philadelphia/src/main/java/org/jvirtanen/philadelphia/FIXConfig.java

```diff
@@ -7,260 +7,311 @@
-
-        /**
-         * Set the HeartBtInt(108).
-         *
-         * @param heartBtInt the HeartBtInt(108)
-         * @return this instance
-         */
-        public Builder setHeartBtInt(int heartBtInt) {
-            this.heartBtInt = heartBtInt;
-
-            return this;
-        }
-
-        /**
-         * Set the incoming MsgSeqNum(34).
-         *
-         * @param incomingMsgSeqNum the incoming MsgSeqNum(34)
-         * @return this instance
-         */
-        public Builder setIncomingMsgSeqNum(long incomingMsgSeqNum) {
-            this.incomingMsgSeqNum = incomingMsgSeqNum;
-
-            return this;
-        }
-
-        /**
-         * Set the outgoing MsgSeqNum(34).
-         *
-         * @param outgoingMsgSeqNum the outgoing MsgSeqNum(34)
-         * @return this instance
-         */
-        public Builder setOutgoingMsgSeqNum(long outgoingMsgSeqNum) {
-            this.outgoingMsgSeqNum = outgoingMsgSeqNum;
-
-            return this;
-        }
-
-        /**
-         * Set the maximum number of fields in a message.
-         *
-         * @param maxFieldCount the maximum number of fields in a message
-         * @return this instance
-         */
-        public Builder setMaxFieldCount(int maxFieldCount) {
-            this.maxFieldCount = maxFieldCount;
-
-            return this;
-        }
-
-        /**
-         * Set the field capacity.
-         *
-         * @param fieldCapacity the field capacity
-         * @return this instance
-         */
-        public Builder setFieldCapacity(int fieldCapacity) {
-            this.fieldCapacity = fieldCapacity;
-
-            return this;
-        }
-
-        /**
-         * Build the session configuration.
-         *
-         * @return the session configuration
-         */
-        public FIXConfig build() {
-            return new FIXConfig(version, senderCompId, targetCompId,
-                    heartBtInt, incomingMsgSeqNum, outgoingMsgSeqNum,
-                    maxFieldCount, fieldCapacity);
-        }
-
-    }
+	private FIXVersion version;
+	private String     senderCompId;
+	private String     targetCompId;
+	private int        heartBtInt;
+	private long       incomingMsgSeqNum;
+	private long       outgoingMsgSeqNum;
+	private int        maxFieldCount;
+	private int        fieldCapacity;
+	private int        rxBufferCapacity;
+	private int        txBufferCapacity;
+
+	/**
+	 * Create a session configuration.
+	 *
+	 * @param version the protocol version
+	 * @param senderCompId the SenderCompID(49)
+	 * @param targetCompId the TargetCompID(56)
+	 * @param heartBtInt the HeartBtInt(108)
+	 * @param incomingMsgSeqNum the incoming MsgSeqNum(34)
+	 * @param outgoingMsgSeqNum the outgoing MsgSeqNum(34)
+	 * @param maxFieldCount the maximum number of fields in a message
+	 * @param fieldCapacity the field capacity
+	 * @param rxBufferCapacity the field receive buffer capacity
+	 * @param txBufferCapacity the field transmit buffer capacity
```

Comment:
These should probably say "the receive buffer capacity" and "the transmit buffer capacity".